R/boot_ts_array.R

```diff
@@ -0,0 +1,35 @@
+#' Copy of boot:::ts.array
+#'
+#' The function `ts.array` of the `boot` package is not exported, yet we depend
+#' on it. It is bad to depend on private functions of a library, so we copy it
+#' here in the current version such that the `boot` package is free to change
+#' it.
+boot_ts_array <- function (n, n.sim, R, l, sim, endcorr) 
```

Comment:
@martin-ueding 
Sorry you're going to quickly on this. There's too much going on at the same time to follow this in full detail. I at least need two or three weeks, not a few hours. Did you check that this works correctly? I think the RNG state needs to be reset for this to give the correct indices in the "geom" case.

There is an exported wrapper for this available: ```boot.array```. I don't think there's any need for using the unexported function.

Reply:
@urbach As a result, I'm not even sure whether your use was correct.

Reply:
I guess that is actually taken care of. However, what happens when the implementation of ```tsboot``` does change how "geom" is implemented? Then this will break silently...

Reply:
I believe that I did not change the working of the code, I just copied `boot:::ts.array` to `boot_ts_array` and change the function calls. This way we are isolated against internal changes in `boot`, nothing more.

Reply:
No, if the internal implementation in ```boot``` changes, this function will give wrong results...